GitCommands/Repository/RepositoryShortNameProvider.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.IO;
+using System.Linq;
+using GitCommands.Git;
+
+namespace GitCommands.Repository
+{
+    public interface IRepositoryShortNameProvider
+    {
+        /// <summary>
+        /// Returns a short name for repository.
+        /// If the repository contains a description it is returned,
+        /// otherwise the last part of path is returned.
+        /// </summary>
+        /// <param name="repositoryDir">Path to repository.</param>
+        /// <returns>Short name for repository</returns>
+        string Get(string repositoryDir);
+    }
+
+    public sealed class RepositoryShortNameProvider : IRepositoryShortNameProvider
+    {
+        private const string RepositoryDescriptionFileName = "description";
+        private const string DefaultDescription = "Unnamed repository; edit this file 'description' to name the repository.";
+        private readonly IGitDirectoryResolver _gitDirectoryResolver;
+
+
+        public RepositoryShortNameProvider(IGitDirectoryResolver gitDirectoryResolver)
+        {
+            _gitDirectoryResolver = gitDirectoryResolver;
+        }
+
+
+        /// <summary>
+        /// Returns a short name for repository.
+        /// If the repository contains a description it is returned,
+        /// otherwise the last part of path is returned.
+        /// </summary>
+        /// <param name="repositoryDir">Path to repository.</param>
+        /// <returns>Short name for repository</returns>
+        public string Get(string repositoryDir)
+        {
+            var dirInfo = new DirectoryInfo(repositoryDir);
+            if (!dirInfo.Exists)
+            {
+                return dirInfo.Name;
```

Comment:
?

Reply:
[As per MSDN](https://msdn.microsoft.com/en-us/library/system.io.directoryinfo.directoryinfo(v=vs.110).aspx#Anchor_2) _This constructor does not check if a directory exists. _

So if the supplied path is invalid or not available (repo on a removable or network drive) you still want to see the name for it.